### PR TITLE
[pipelines] Fix panic in KPI parsing

### PR DESCRIPTION
**Pull request by drewrip**
Explicitly handle when there is incomplete Run duration or Run latency data provided.

### CODE REVIEW

**Comment by drewrip** [comment]
@kpouget Both the default [test now passes](https://ci.app-svc-perf.corp.redhat.com/job/ExternalTeams/job/RHODS/job/topsail/1439/artifact/run/f23-h33-000-6018r.rdu2.scalelab.redhat.com//002_test_ci), and when running with `wait_for_run_completion` [passes](https://ci.app-svc-perf.corp.redhat.com/job/ExternalTeams/job/RHODS/job/topsail/1440/artifact/run/f23-h33-000-6018r.rdu2.scalelab.redhat.com//002_test_ci). Good to merge from my end

File: projects/pipelines/visualizations/rhods-pipelines/store/lts_parser.py [hunk 1]
```diff
@@ -134,26 +139,48 @@ def _generate_run_duration(results):
...
-    test_stats["max"] = np.max(data)
-    test_stats["median"] = np.median(data)
-    test_stats["mean"] = np.mean(data)
-    test_stats["percentile_80"] = np.percentile(data, 80)
-    test_stats["percentile_90"] = np.percentile(data, 90)
-    test_stats["percentile_95"] = np.percentile(data, 95)
-    test_stats["percentile_99"] = np.percentile(data, 99)
+    # Some of the Numpy functions panic on 0-length arrays
+    # so we need to check this explicitly
+    if len(data) <= 0:
```
Comment (kpouget) [inline projects/pipelines/visualizations/rhods-pipelines/store/lts_parser.py:159]:
```suggestion
    if not data:
```

File: projects/pipelines/visualizations/rhods-pipelines/store/lts_parser.py [hunk 2]
```diff
@@ -134,26 +139,48 @@ def _generate_run_duration(results):
...
                     data[resource_key].append((user_data.complete_run_times[workflow_run_name] - user_data.workflow_start_times[resource_name.split("/")[1]]).total_seconds())
 
-    run_duration = _generate_dsp_test_stats(list(reduce(lambda l, r: l + r, data.values(), [])))
-    medians = np.array([np.median(d[1]) for d in sorted(data.items())])
-    A = np.vstack([np.arange(len(medians)), np.ones(len(medians))]).T
-    slope, _ = np.linalg.lstsq(A, medians, rcond=None)[0]
-    run_duration["degrade_speed"] = slope
+    all_points = list(reduce(lambda l, r: l + r, data.values(), []))
+    run_duration = _generate_dsp_test_stats(all_points)
+    if len(all_points) <= 0:
```
Comment (kpouget) [inline projects/pipelines/visualizations/rhods-pipelines/store/lts_parser.py:144]:
```suggestion
    if not all_points:
```

File: projects/pipelines/visualizations/rhods-pipelines/store/lts_parser.py (hunk 1 above)
Comment (drewrip) [reply projects/pipelines/visualizations/rhods-pipelines/store/lts_parser.py:159]: Personally I don't like relying on the implicit bool conversion in Python. Happy to make the switch in the name of consistency though.

Comment (kpouget) [reply projects/pipelines/visualizations/rhods-pipelines/store/lts_parser.py:159]: yes, please update if you don't mind

**Comment by kpouget** [comment]
great, thanks Drew, merging!

**Pull request merged**